### PR TITLE
Client side validation of crop add/edit form

### LINKED ISSUE #806: Create/edit crop UI does not prevent invalid form data submission
1. Create new crop
2. Enter a spacing of '25-50' cm (which is invalid)
3. Submit form

Expected:
I should have been prevented from entering invalid data or submitting the form

Actual:
Silently discarded invalid data, form submitted

The UI is applying classes to indicate the field is invalid; but it isn't styled.

image

### PR DESCRIPTION
**Pull request by speizerj**
Resolves #806.

By adding the `required` attribute to any required fields (such as crop name), the form will now make sure the required fields are present and if not, will disable the submit button. 

It will also be disabled if any of the fields have the `ng-invalid` class applied to it, which happens when the number inputs are invalid. The invalid input forms will then have a red border applied to it to alert the user of the reason why they can't submit the form, but we probably want to have some kind of form messaging system that makes it clearer.

screenshot 2016-10-04 16 00 49

screenshot 2016-10-04 16 00 42

### CODE REVIEW

File: app/assets/stylesheets/styles/ui/inputs/inputs.scss [hunk 1]
```diff
@@ -15,3 +15,7 @@ input[type="color"],
 textarea {
   border-radius: $global-radius;
 }
+
+input.ng-invalid {
+  border: 1px solid red;
```
Comment (simonv3) [inline app/assets/stylesheets/styles/ui/inputs/inputs.scss]: I think (from memory) there's an $of-red variable (check _variables.scss). There might even be an $of-alert - or something similar - color.

Comment (speizerj) [inline app/assets/stylesheets/styles/ui/inputs/inputs.scss]: Yeah, good call - using $alert-color

**Comment by simonv3** [comment]
This is great, thanks @speizerj! I'll let TravisCI run its course and then merge it (I'm sure it'll pass). 

I'll also add you as a core contributor to the project :). We've had that as a policy for a while, but hadn't yet specified it in the Readme (which I've [now fixed[(https://github.com/openfarmcc/OpenFarm#core-contributors)).

**Comment by speizerj** [comment]
Awesome, thanks @simonv3! Really looking forward to contributing a lot to the project

**Pull request merged**